File: frontend/src/mocks/alunoCnhStatus.ts
import { AlunoCnhStatus } from '@/types';
import { mockUsuarios } from './usuarios';
import { mockAulasPraticas } from './aulasPraticas';

export const mockAlunoCnhStatus: AlunoCnhStatus = {
    id: 1,
    usuarioId: "usr-001",
    usuario: mockUsuarios[0],
    possuiContaGov: true,
    processoIniciadoDetran: true,
    exameMedicoRealizado: true,
    exameMedicoAprovado: true,
    caminhoExameMedico: "/docs/exame-medico.pdf",
    caminhoPsicotecnico: "/docs/psicotecnico.pdf",
    dataEnvioExames: "2024-09-01T14:00:00",
    examesEnviados: true,
    exameTeoricoRealizado: true,
    exameTeoricoAprovado: true,
    caminhoExameTeorico: "/docs/exame-teorico.pdf",
    dataEnvioExameTeorico: "2024-10-15T10:00:00",
    aulasPraticasIniciadas: true,
    aulasPraticas: mockAulasPraticas,
    primeiroAcesso: false,
    ultimaAtualizacao: "2025-02-17T08:00:00"
};

File: Cnh_rapida/Models/AlunoCnhStatus.cs
namespace Cnh_rapida.Models;

public class AlunoCnhStatus
{
    public int Id { get; set; }

    public string UsuarioId { get; set; } = string.Empty;
    public virtual Usuario Usuario { get; set; } = null!;

    public int? AutoEscolaId { get; set; }
    public virtual AutoEscola? AutoEscola { get; set; }

    public bool PossuiContaGov { get; set; } = false;
    public bool ProcessoIniciadoDetran { get; set; } = false;

    public bool ExameMedicoRealizado { get; set; } = false;
    public bool ExameTeoricoRealizado { get; set; }
    public bool AulasPraticasIniciadas { get; set; }

    public bool PrimeiroAcesso { get; set; } = true;

    // 🔹 CONTROLE DO UPLOAD
    public bool ExamesEnviados { get; set; } = false;
    public bool ExameMedicoAprovado { get; set; } = false;

    public string? CaminhoExameMedico { get; set; }
    public string? CaminhoPsicotecnico { get; set; }

    public DateTime? DataEnvioExames { get; set; }

    public bool ExameTeoricoAprovado { get; set; } = false;
    public string? CaminhoExameTeorico { get; set; }
    public DateTime? DataEnvioExameTeorico { get; set; }

    public bool DocumentosAprovados { get; set; } = false;

    public DateTime UltimaAtualizacao { get; set; } = DateTime.UtcNow;

    public virtual ICollection<AulaPratica> AulasPraticas { get; set; }
    = new List<AulaPratica>();

}
